### DISCUSSION TITLE
Stuck trying to clone GitHub repo to local

### DISCUSSION THREAD

**Discussion by mesirhc** [community/community#53898] [New to GitHub] [answered]
Hi Everyone,

I'm in my first few days of GitHub & Git ...I think I've got a handle on running Git locally on my Mac, but GitHub has defeated me.

The project is: Private(for now),100% C, me as only dev, GPLv3: an edge-case in Constraint Programming.
The source (around 50k in 14 .c & .h files) is up on GitHub in branch: main.

Here is a (redacted) transcript of my my clone attempt:
```
user@mac dev % git clone https://github.com/pseudo/abc.git
Cloning into 'abc'...
Username for 'https://github.com': pseudo
Password for 'https://pseudo@github.com': 
remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/pseudo/abc.git/'
user@mac dev %
```

Unfortunately, five hours digging in to those and other docs and this forum has not shed any light!

In case it changes anything, my dev environment is:
	M1 Mac-Air Ventura 13 3.1 , 
	BBEdit  14.6.5 (14D70, Apple Silicon, sandboxed)
	Clang 1403.0.22.14.1 
	Git 2.39.2 (Apple git 143).

Suggestions please :)

ChrisM

**Comment by airtower-luna** [comment, marked as answer]
As the message says, you cannot use your password for `git push` or other Git operations (but you still need it to log in). You need to [create a personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) and enter that instead when Git asks for your password. When creating the token make sure to select the **repo** scope (for repository access, like pushing) and if you use Actions also **workflow** (required to modify workflow files).

**Reply by mesirhc** [reply]
Sorry for the delayed reply: eating!
I'll give the "personal access token" route a go now.

Thanks for the suggestion. ChrisM

**Reply by mesirhc** [reply]
@airtower-luna 

I had tried before your reply, I tried again a few moments ago:
Created a new PAT for Contents.
Received:
`github_pat_227H_THIS_IS_REDACTED_89QZ33ii_jj_kk_ll_mm_yada_yada_yada_blah_etc`

I 'm the proud owner of yet another github_pat.
Now what to do with it?

I got there in the end with:

```
$ git clone https://github.com/pseudo/abc.git
Username: chrism@example.foo
Password: github_pat_227H_THIS_IS_REDACTED_89QZ33ii_jj_kk_ll_mm_yada_yada_yada_blah_etc
```

Thanks for pushing me to access tokens.

I learnt:
	USER_NAME != github pseudo // use mail-address
	pat has an annoying expiry date

Onwards, upwards!

Thanks ChrisM

**Reply by airtower-luna** [reply]
Your username should actually work, but with a token it doesn't matter either way. :wink:

**Reply by mesirhc** [reply]
It is of no consequence IFF my mail-address will be accepted "GitHub-everywhere" as 'Username'.
I'll try again tomorrow morning using my GitHub pseudo just for the fun of it (perhaps a typo in my prior attempts?).

...imbibed rather too much wine to play code-games tonight tho.

ChrisM

**Reply by airtower-luna** [reply]
Oh yes, "don't drink and code" is good advice. :grin:

**Reply by Docdewy** [reply]
Should identify themself with own account, another thing I've got to deal with.

**Reply by Docdewy** [reply]
My Apologies!

**Reply by Docdewy** [reply]
This hacker has overided my email now and probably already had you believing he is me
 Since he's running my phone and has me as client/guest on my phone
